src/checks/stylelint.ts

```diff
@@ -0,0 +1,65 @@
+import { execaCommand } from 'execa';
+import { LintResult } from 'stylelint';
+import { projectPackageJson } from '../util/packageJson';
+
+const projectDependencies = projectPackageJson.dependencies || {};
+
+export const supportedFileExtensions = [
+  'css',
+  'sass',
+  'scss',
+  'less',
+  'js',
+  'tsx',
+  'jsx',
+];
+
+export const title = 'Stylelint';
+
+export default async function stylelintCheck() {
+  // Continue only if project is Next.js or UpLeveled React App
+  if (
+    !(
+      '@upleveled/react-scripts' in projectDependencies ||
+      'next' in projectDependencies
+    )
+  ) {
+    return;
+  }
+
+  try {
+    await execaCommand(
+      `yarn stylelint **/*.{${supportedFileExtensions.join(
+        ',',
+      )}} --max-warnings 0 --formatter json`,
+    );
+  } catch (error) {
+    const { stdout } = error as { stdout: string };
+
+    const stylelintJSONOutput = stdout.slice(
+      stdout.indexOf('['),
+      stdout.lastIndexOf(']') + 1,
+    );
```

Comment:
- `stylelintJsonOutput` (lowercase `son` in `Json`) - camelcase is more important than casing within a word
- what is this doing? can we do this in a simpler way? if there is absolutely no simpler way we should probably comment this and show the data structure

Reply:
1c9b999

Reply:
2807a58